chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/ListingSvapActivityHistoryHelper.java

```diff
@@ -0,0 +1,84 @@
+package gov.healthit.chpl.activity.history;
+
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import gov.healthit.chpl.dao.ActivityDAO;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.activity.ActivityConcept;
+import gov.healthit.chpl.dto.ActivityDTO;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+public class ListingSvapActivityHistoryHelper {
```

Comment:
I'm not a big fan of classes ending with "Helper".  I know I have added several classes to CHPL with this suffix, so you can take this with a grain of salt.  It is often considered a code smell.  This class looks like it get Activity based on some type of criteria (not a CHPL "Criteria"), and will/can be expanded in the future to support other criteria.  Maybe something like `ListingActivityHistoryEvent` - I'm not sure...  Naming is hard!!

Or maybe make a `ListingActivityHistory` interface.  Then make an implementation of 'LastSvapNoticeUrlListingActivityHistory` .  In this model, each "query would be it's own class, but we might have more control in the long run, by being able to run several of these "queries" in a generic way.